api.json

```diff
@@ -0,0 +1,802 @@
+{
+    "openapi": "3.0.1",
+    "info": {
+        "title": "Deku",
+        "version": "v1",
+        "description": "API for deku v1."
+    },
+    "paths": {
+        "/chain/info": {
+            "get": {
+                "description": "Returns some informations about the chain: the consensus and discovery contract address.",
+                "tags": [
+                    "chain"
+                ],
+                "responses": {
+                    "200": {
+                        "description": "consensus and discovery address",
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "$ref": "#/components/schemas/ChainInfo"
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "/chain/level": {
+            "get": {
+                "description": "Returns the current level of the chain.",
+                "tags": [
+                    "chain"
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Returns the current level of the chain as a string representing an int64.",
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "type": "object",
+                                    "properties": {
+                                        "level": {
+                                            "$ref": "#/components/schemas/BlockLevel"
+                                        }
+                                    }
+                                },
+                                "examples": {
+                                    "level": {
+                                        "$ref": "#/components/examples/level-response"
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "/chain/blocks": {
+            "get": {
+                "description": "Streams blocks starting at the current level of the chain",
```

Comment:
Since this endpoint only streams from the actual moment you call it, will it be possible to retrieve the previous blocks?
Maybe adding a parameter like `starting_block` could be great? Which will make this endpoint to stream from a given point.
This is a naive question, I don't know if it is relevant to do so.

Reply:
It's complicated to answer this question, should the search of old blocks be part of the chain or an indexer ? 
This endpoint is only used to observe when an operation is applied:
 - you send an operation
 - listen to block
 - until you see your operation in a block
 - you know the status of your operation (accepted or not)
This is not the best way to know the status of your operation, and maybe in the future we will add an endpoint to know the status of your operation

Reply:
What if no indexer is running when the chain starts? Are those blocks lost?
I think we want a notion of a "archive node" that just hold all the blocks and allows you to query them.